Packages/io.chainsafe.web3-unity/Runtime/Scripts/ChainSafeManager.cs

```diff
@@ -0,0 +1,150 @@
+        
+        public static ChainSafeManager Instance
+        {
+            get
+            {
+                if (!_instance)
+                {
+                    _instance = FindObjectOfType<ChainSafeManager>();
+                }
+                
+                return _instance;
+            }
+
+            private set => _instance = value;
+        }
+
+        public static CWeb3 Web3 => Instance._web3;
+        
+        public string Address => Web3?.Signer.PublicAddress;
+        
+        public bool Connected { get; private set; }
+        
+        [SerializeField] private bool connectOnInitialize = true;
+        
+        [DefaultAssetValue("Packages/io.chainsafe.web3-unity/Runtime/Prefabs/Connect.prefab")]
+        [SerializeField] private ConnectModal connectModalPrefab;
+
+        private CWeb3 _web3;
+        
+        private ConnectionHandler _connectionHandler;
+        
+        private ConnectModal _connectModal;
+
+        private void Awake()
+        {
+            DontDestroyOnLoad(gameObject);
+        }
+
+        public async Task Initialize()
+        {
+            _connectionHandler = FindObjectOfType<ConnectionHandler>();
+            
+            await _connectionHandler.Initialize();
+
+            try
+            {
+                if (connectOnInitialize)
+                {
+                    _web3 = await _connectionHandler.Restore();
+                }
+            }
+            finally
+            {
+                if (!Connected)
+                {
+                    _connectModal = Instantiate(connectModalPrefab);
+                    
+                    _connectModal.Initialize(_connectionHandler.Providers);
+                }
+            }
+        }
+
+        public async Task Connect(ConnectionProvider provider)
+        {
+            _web3 = await (_connectionHandler as IConnectionHandler).Connect(provider);
+
+            if (Web3 != null)
+            {
+                Connected = true;
+            }
+        }
+        
+        public async Task Connect<T>() where T : ConnectionProvider
+        {
+            if (!_connectionHandler.GetProvider(out T provider))
+            {
+                throw new Web3Exception($"{typeof(T).Name} unavailable or disabled.");
+            }
+            
+            await Connect(provider);
+        }
+
+        public void ShowModal()
+        {
+            _connectModal.Show();
+        }
+        
+        public Task<string> SignMessage(string message)
+        {
+            AssetConnection();
+            
+            return Web3.Signer.SignMessage(message);
+        }
+
+        public Task<string> SignTypedData<TStructType>(SerializableDomain domain, TStructType message)
+        {
+            AssetConnection();
+            
+            return Web3.Signer.SignTypedData(domain, message);
+        }
+        
+        public Task<TransactionResponse> SendTransaction(TransactionRequest transaction)
```

Comment:
I would ommit this. Since we have ABI C# converter, no need for having SendTransaction in here.